openstack/blockstorage/v2/volumes/requests.go

```diff
@@ -122,6 +122,36 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 	})
 }
 
+// ExtendOptsBuilder allows extensions to add additional parameters to the
+// extend action (os-extend) request.
+type ExtendOptsBuilder interface {
+	ToVolumeExtendMap() (map[string]interface{}, error)
+}
+
+// ExtendOpts holds options for extending volumes. It is passed to the volumes.Extend
+// function.
+type ExtendOpts struct {
+	//New size of the volume in GiB
+	NewSize int `json:"new_size" required:"true"`
+}
+
+func (opts ExtendOpts) ToVolumeExtendMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "os-extend")
+}
+
+// Extend will extend the Volume with specified volume size.
+func Extend(client *gophercloud.ServiceClient, id string, opts ExtendOptsBuilder) (r ActionResult) {
```

Comment:
I'd be curious what others think, but it also might be worth considering for cases like this (functions that have only one or two args) to just take a variable directly.  So something like:
    func Extend(client *gophercloud.Serviceclient, id string, newSize int64)

I realize there's a question of consistency, but just throwing it out there as it seems reasonable.